### DISCUSSION TITLE
Debug Feature - Holding Browser Session

### DISCUSSION THREAD

**Discussion by vijaychowthri** [webdriverio/webdriverio#8458] [Ideas]
**Problem statement:**  we have a web application work flow navigates more than 20 pages, if something fails on the 11th page, the team has to fix and re-run the test multiple times. until it gets fixed. It's time killing process.

is it possible to implement the features that allow users to retain the session and page until fixing the issue?

**Comment by christian-bromann** [comment]
> is it possible to implement the features that allow users to retain the session and page until fixing the issue?

You mean something like snapshotting a browser session and restart from a certain point?

**Reply by vijaychowthri** [reply]
Yes

**Reply by christian-bromann** [reply]
I am not sure if a browser allows this but I would be interested to look into this. At least for Chrome and Firefox it should be possible to zip the user data files, store them and reapply them in a later session. I am not sure if this would also include cookies given they might be stored on a system level. This would also only work for local development and not using cloud vendors.

**Reply by vijaychowthri** [reply]
that would be great !!!

**Comment by vijaychowthri** [comment]
HP &  IBM tools have such a feature, where you can keep re-run your code by updating conditions on certain pages and fixing the scripts.
